### PR TITLE
feat: allow user-defined balancer with metadata in node

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/schema_def.lua [hunk 1]
```diff
@@ -389,7 +389,6 @@ local upstream_schema = {
         type = {
             description = "algorithms of load balancing",
             type = "string",
-            enum = {"chash", "roundrobin", "ewma", "least_conn"}
```
Comment (Yiyiyimu) [inline apisix/schema_def.lua:392]: do we need to test the behaviour when a non-user-defined algorithm is chosen

File: docs/en/latest/admin-api.md [hunk 2]
```diff
@@ -568,6 +568,7 @@ In addition to the basic complex equalization algorithm selection, APISIX's Upst
 * `chash`: consistent hash
 * `ewma`: pick one of node which has minimum latency. See https://en.wikipedia.org/wiki/EWMA_chart for details.
 * `least_conn`: pick node which has the lowest `(active_conn + 1) / weight`. Note the `active connection` concept is the same with Nginx: it is a connection in used by a request.
+* user-defined balancer which can be loaed via `require("apisix.balancer.your_balancer")`.
```
Comment (Yiyiyimu) [inline docs/en/latest/admin-api.md:571]: I failed to find the context so I'm not sure the relationship between "allow user-defined balancer" and "discovery when metadata changed". Are they two separate features?

**Review by Yiyiyimu — APPROVED** [review]

File: docs/en/latest/admin-api.md (hunk 2 above)
Comment (spacewander) [reply docs/en/latest/admin-api.md:571]: The user-defined balancer needs to use the metadata.

**Review by tokers — APPROVED** [review]

File: docs/en/latest/admin-api.md (hunk 2 above)
Comment (Yiyiyimu) [reply docs/en/latest/admin-api.md:571]: Got it. Thanks for explaination

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**